packages/experimental/Expander/src/Expander.tsx

```diff
@@ -21,18 +21,30 @@ export const Expander = compose<ExpanderType>({
   render: (userProps: ExpanderProps, useSlots: UseSlots<ExpanderType>) => {
     const Root = useSlots(userProps).root;
     const [expandedState, setExpandedState] = React.useState(userProps.expanded);
+    const expanderHeight = expandedState? userProps.expandedHeight : userProps.collapsedHeight;
 
-    return (rest: ExpanderViewProps, ...children: React.ReactNode[]) => <Root
-      height= {expandedState? userProps.expandedHeight : userProps.collapsedHeight}
-      onChange={async (event: ExpanderChangeEvent) => {
-        if (event.nativeEvent.expanded != null) {
-          event.persist();
-          if (!event.nativeEvent.expanded) {
-            await delay(175);
-          }
-          setExpandedState(event.nativeEvent.expanded);
-        }
-      }}
-      {...mergeProps(userProps, rest)}>{children}</Root>;
+    const _onCollapsing = async () => {
+      if (userProps.onCollapsing !== undefined) {
+        userProps.onCollapsing();
+      }
+      // Need to delay the height change so that the animation runs
+      await delay(175);
+      setExpandedState(false);
+    }
+
+    const _onExpanding = () => {
+      setExpandedState(true);
+      if (userProps.onExpanding !== undefined) {
+        userProps.onExpanding();
+      }
+    }
+
+    return (rest: ExpanderViewProps, ...children: React.ReactNode[]) =>
+      <Root
+        {...mergeProps(userProps, rest)}
+        height={expanderHeight}
+        onCollapsing={_onCollapsing}
+        onExpanding={_onExpanding}>{children}
```

Comment:
nit move `children` to next line

Reply:
little surprised the linter didn't pick it up 🤔